lambda/shared/src/main/scala/feral/lambda/events/S3BatchEvent.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda.events
+
+import feral.lambda.KernelSource
+import io.circe.Decoder
+
+// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/trigger/s3-batch.d.ts
+
+final case class S3BatchEvent(
+    invocationSchemaVersion: String,
+    invocationId: String,
+    job: S3BatchJob,
+    tasks: List[S3BatchEventTask]
+)
+
+object S3BatchEvent {
+  implicit val decoder: Decoder[S3BatchEvent] =
+    Decoder.forProduct4("invocationSchemaVersion", "invocationId", "job", "tasks")(
+      S3BatchEvent.apply)
+
+  implicit def kernelSource: KernelSource[S3BatchEvent] = KernelSource.emptyKernelSource
+}
+
+final case class S3BatchJob(id: String)
```

Comment:
Seems it's called `S3BatchEventJob` in TS :)
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d18a30127146217038c61060b06e96713daa706c/types/aws-lambda/trigger/s3-batch.d.ts#L17